lib/Drush/Psysh/DrushCommand.php

```diff
@@ -0,0 +1,227 @@
+      ->setDefinition($this->buildDefinitionFromConfig())
+      ->setDescription($this->config['description'])
+      ->setHelp($this->buildHelpFromConfig());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $args = $input->getArguments();
+    $first = array_shift($args);
+
+    // If the first argument is an alias, assign the next argument as the
+    // command.
+    if (strpos($first, '@') === 0) {
+      $alias = $first;
+      $command = array_shift($args);
+    }
+    // Otherwise, default the alias to '@self' and use the first argument as the
+    // command.
+    else {
+      $alias = '@self';
+      $command = $first;
+    }
+
+    $return = drush_invoke_process($alias, $command, array_values($args), $input->getOptions(), ['backend' => TRUE]);
+
+    if ($return['error_status'] > 0) {
+      foreach ($return['error_log'] as $error_type => $errors) {
+        $output->write($errors);
+      }
+      // Add a newline after so the shell returns on a new line.
+      $output->writeln('');
+    }
+    else {
+      $output->page(drush_backend_get_result());
+    }
+  }
+
+  /**
+   * Extract drush command aliases from config array.
+   *
+   * @return array
+   *   The command aliases.
+   */
+  protected function buildAliasesFromConfig() {
+    return !empty($this->config['aliases']) ? $this->config['aliases'] : [];
+  }
+
+  /**
+   * Build a command definition from drush command configuration array.
+   *
+   * Currently, adds all non-hidden arguments and options, and makes a decent
+   * effort to guess whether an option accepts a value or not. It isn't always
+   * right :P
+   *
+   * @return array
+   *   the command definition.
+   */
+  protected function buildDefinitionFromConfig() {
+    $definitions = [];
+
+    if (isset($this->config['arguments']) && !empty($this->config['arguments'])) {
+      $required_args = $this->config['required-arguments'];
+
+      if ($required_args === FALSE) {
+        $required_args = 0;
+      }
+      elseif ($required_args === TRUE) {
+        $required_args = count($this->config['arguments']);
+      }
+
+      foreach ($this->config['arguments'] as $name => $argument) {
+        if (!is_array($argument)) {
+          $argument = ['description' => $argument];
+        }
+
+        if (!empty($argument['hidden'])) {
+          continue;
+        }
+
+        $input_type = ($required_args-- > 0) ? InputArgument::REQUIRED : InputArgument::OPTIONAL;
+
+        $definitions[] = new InputArgument($name, $input_type, $argument['description'], NULL);
+      }
+    }
+
+    if (!empty($this->config['options'])) {
+      foreach ($this->config['options'] as $name => $option) {
+        if (!is_array($option)) {
+          $option = ['description' => $option];
+        }
+
+        if (!empty($option['hidden'])) {
+          continue;
+        }
+
+        // @todo: Figure out if there's a way to detect InputOption::VALUE_NONE
```

Comment:
Spoke to @weitzman about this one, there is indeed no reliable way to do this currently. Let's punt this for a follow up.